lib/select.c

```diff
@@ -216,11 +219,15 @@ int Curl_socket_check(curl_socket_t readfd0, /* two sockets to read from */
   int r;
   int ret;
 
-#if SIZEOF_TIME_T != SIZEOF_INT
-  /* wrap-around precaution */
-  if(timeout_ms >= INT_MAX)
+  /* prevent overflow. timeout_ms is typecast to time_t and int. */
+#if TIMEDIFF_T_MAX > INT_MAX
+  if(timeout_ms > INT_MAX)
     timeout_ms = INT_MAX;
 #endif
+#if INT_MAX > TIME_T_MAX
+  if(timeout_ms > (int)TIME_T_MAX)
+    timeout_ms = (int)TIME_T_MAX;
+#endif
```

Comment:
Can you mention a platform where this change is necessary?

Reply:
time_t unsigned

Reply:
While it might be correct to check, curl doesn't build on any platform where `int` is not 32 bit. I'm sure such a theoretical platform would break the curl build in numerous other places. And @mback2k has a larger take on this code in progress anyway. But I won't object if you insist.

Reply:
Yes I can't think of one. For posterity. #5262 may cover this PR, and if not it can be incorporated. I'll wait to see what happens there.

Reply:
@jay I would be fine with merging this PR before the release, as my changes will be part of the next feature window.

Reply:
Ok. I will put this in for now and then you can replace it when you're ready.